### PR TITLE
JVM-1722: add PEA_debug_idx to select one ObjID

### PR DESCRIPTION
**Pull request by navyxliu**
Add a debuggability option PEA_debug_idx. when it's > 0, PEA only tracks ObjID == PEA_debug_idx.  if PEA_debug_idx < 0, PEA skip it

### CODE REVIEW

File: src/hotspot/share/opto/parseHelper.cpp [hunk 1]
```diff
@@ -358,6 +358,15 @@ void PEAState::add_new_allocation(Node* obj) {
 
   if (nfields >= 0) {
     AllocateNode* alloc = obj->in(1)->in(0)->as_Allocate();
+#ifndef PRODUCT
+    if (PEA_debug_idx != 0) {
```
Comment (caojoshua) [inline src/hotspot/share/opto/parseHelper.cpp]: looks like this if statement is not required. Checks covered by nested if-else.

Comment (navyxliu) [reply src/hotspot/share/opto/parseHelper.cpp]: yes. it's left-over code.

File: src/hotspot/share/opto/parseHelper.cpp [hunk 2]
```diff
@@ -358,6 +358,13 @@ void PEAState::add_new_allocation(Node* obj) {
 
   if (nfields >= 0) {
     AllocateNode* alloc = obj->in(1)->in(0)->as_Allocate();
+#ifndef PRODUCT
+    if (PEA_debug_idx > 0 && alloc->_idx != PEA_debug_idx) {         // only allow PEA_debug_idx
+        return;
```
Comment (caojoshua) [inline src/hotspot/share/opto/parseHelper.cpp]: NIT: inconsistent indent.

**Review by caojoshua — APPROVED** [review]

File: src/hotspot/share/opto/parseHelper.cpp (hunk 2 above)
Comment (navyxliu) [reply src/hotspot/share/opto/parseHelper.cpp]: I see. I will update it after I confirm that it can compile on x86-32.

**Pull request merged**